### PR TITLE
Basic Acceptance Test for Echo Server

### PR DESCRIPTION
**Pull request by olgaboiar**
- Added Cucumber to the project
- Added Feature description
- Added Steps definitions
- Refactored code until tests pass

### CODE REVIEW

File: src/main/java/com/olgaboiar/guacomole/ServerRunner.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.olgaboiar.guacomole;
+
+import java.io.IOException;
+
+public class ServerRunner {
+    public static void main(String[] args) throws IOException {
+        EchoServer server;
+        server = new EchoServer();
+        server.start("localhost", 3333);
```
Comment (naush) [inline src/main/java/com/olgaboiar/guacomole/ServerRunner.java]: Can we make the host and port passed in through command line arguments?

File: src/test/java/gradle/cucumber/ServerProcess.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+public class ServerProcess {
+    Process serverProcess;
+
+    public static ServerProcess start() throws IOException{
+        return new ServerProcess();
+    }
+
+    private ServerProcess() throws IOException {
+        serverProcess = new ProcessBuilder()
+                .command("java", "-jar", "build/libs/com.olgaboiar.guacomole-1.0-SNAPSHOT.jar")
```
Comment (naush) [inline src/test/java/gradle/cucumber/ServerProcess.java]: Can we make the location of the executable configurable?

File: src/main/java/com/olgaboiar/guacomole/ServerRunner.java (hunk 1 above)
Comment (olgaboiar) [reply src/main/java/com/olgaboiar/guacomole/ServerRunner.java]: implemented in https://github.com/olgaboiar/GuacamoleServer/pull/2/commits/f8ca23456d33f499d56b2a558f170126593b5df3

File: src/test/java/gradle/cucumber/ServerProcess.java (hunk 2 above)
Comment (olgaboiar) [reply src/test/java/gradle/cucumber/ServerProcess.java]: implemented in https://github.com/olgaboiar/GuacamoleServer/pull/2/commits/4155dd3e702a428653f375a6b052f1b246ef684e

**Pull request merged**